src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -364,7 +356,18 @@ public MultipleMutationInputValidator(IMetadataProviderFactory sqlMetadataProvid
                 metadataProvider: metadataProvider,
                 columnDataInSourceBody: backingColumnData,
                 targetNodeValue: relationshipFieldValue,
-                nestingLevel: nestingLevel);
+                nestingLevel: nestingLevel,
+                isMNRelationship: isMNRelationship);
+
+            if (isMNRelationship)
+            {
+                // The presence of a linking object indicates that an M:N relationship exists between the current entity and the target/child entity.
+                // The linking table acts as a referencing table for both the source/target entities which act as
+                // referenced entities. Consequently:
+                // - Column values for the target entity can't be derived from insertion in the current entity.
+                // - Column values for the current entity can't be derived from the insertion in the target/child entity.
+                return;
+            }
```

Comment:
this check for `isMNRelationship` should occur before the call to `MultipleCreateOrderHelper.GetReferencingEntityName(...)` so that you can return early.